apps/homework/tasks.py

```diff
@@ -53,29 +65,30 @@ def post_submission(submission_pk, data_file=None):
                 else:
                     new_path += component
             repo_url = '{scheme}://{domain}{path}'.format(scheme=repo_scheme, domain=repo_loc, path=new_path)
+            logger.info(f"Streaming github file for submission: {submission.pk}.")
             repo_resp = requests.get(repo_url, stream=True)
             for chunk in repo_resp.iter_content(chunk_size=1024):
                 if chunk:  # filter out keep-alive new chunks
                     f.write(chunk)
             temp_size = str(f.tell())
             f.seek(0)
-            storage_resp = requests.put(
-                url=s3_file_url,
-                data=f,
-                headers={
-                    "x-ms-blob-type": 'BlockBlob',
-                    "Content-Length": temp_size,
-                }
-            )
+            data_to_upload = f
+            logger.info(f"Succesfully read file from Github for submission: {submission.pk}")
         else:
+            data_to_upload = data_file
+        if data_to_upload:
             storage_resp = requests.put(
                 url=s3_file_url,
-                data=data_file,
+                data=data_to_upload,
                 headers={
                     "x-ms-blob-type": 'BlockBlob',
-                    "Content-Length": str(data_file.size),
+                    "Content-Length": str(data_to_upload.size) if data_file else temp_size,
```

Comment:
should we just send `temp_size` all the time here to keep this simple?

Reply:
👍